### DISCUSSION TITLE
Username suddenly not found?

### DISCUSSION THREAD

**Discussion by ToothyDev** [anuraghazra/github-readme-stats#2668] [Q&A] [answered]
I had this set up for quite some time with wakatime and it worked just fine. However, when I checked today, it broke, saying "Wakatime user not found". I have changed absolutely nothing about my wakatime nor the link in my profile, here is the link I used:
https://github-readme-stats.vercel.app/api/wakatime?username=ToothyDev&theme=radical&hide_border=true&show_icons=True&layout=compact

And here is my wakatime link: https://wakatime.com/ToothyDev

**Comment by qwerty541** [comment]
Check this discussion: #1825
I hope it will help you

**Comment by ToothyDev** [comment]
My profile is as public as it goes and more than 4h have passed, it still doesn't show

**Comment by qwerty541** [comment]
I see that a lot of users currently have same issue. Most likely, this does not depend on you and no action is required from you. I think that merging #2670 will resolve that,

**Reply by qwerty541** [reply, marked as answer]
Or you can manually set `&range=all_time` option to fix it now.

![](https://github-readme-stats.vercel.app/api/wakatime?username=ToothyDev&theme=radical&hide_border=true&show_icons=True&layout=compact&range=all_time)

**Reply by rickstaa** [reply]
It should be fixed now that #2662 was merged.

**Comment by AmodeusR** [comment]
I'm facing the same issue. The "&range=all_time" didn't work for me. It seems I'll need to wait for the fix.

**Reply by rickstaa** [reply]
It should be fixed now that #2662 was merged.

**Reply by AmodeusR** [reply]
My profile is still showing the error.
![image](https://user-images.githubusercontent.com/29683600/236838823-aad9b83c-b48a-4d93-8293-d3c20883850f.png)

**Reply by nemchik** [reply]
Make sure you have `Display code time publicly` set to `All Time` at https://wakatime.com/settings/profile

If you prefer your image to display a shorter range, you can use `&range=last_year` or `&range=last_7_days` but the profile setting has to be at least enough to include the `range` you want to use. Personally I leave the profile setting on `All Time` and use the image with `&range=last_7_days`.

**Reply by rickstaa** [reply]
@AmodeusR the problem still wasn't fixed. Will be after https://github.com/anuraghazra/github-readme-stats/pull/2707 is merged.

**Comment by hakula139** [comment]
According to the [doc](https://github.com/anuraghazra/github-readme-stats/blob/master/readme.md#wakatime-card-exclusive-options):

> `range` – Request a range different from your WakaTime default, e.g. `last_7_days`. See [WakaTime API docs](https://wakatime.com/developers#stats) for a list of available options. _(YYYY-MM, last_7_days, last_30_days, last_6_months, last_year, or all_time)_. Default `all_time`.

And I've found if I set a range the SAME as my WakaTime settings [here](https://wakatime.com/settings/profile) (Display code time publicly), e.g. `&range=last_7_days` (instead of `&range=all_time`), it solves my issue.

**Reply by rickstaa** [reply]
It should be fixed now that #2662 was merged.

**Reply by hakula139** [reply]
> It should be fixed now that #2662 was merged.

This PR is not related to this issue, as you may see the problem still exists [here](https://github-readme-stats.vercel.app/api/wakatime?username=Hakula), but fixed by adding a `range` param like [this](https://github-readme-stats.vercel.app/api/wakatime?username=Hakula&range=last_7_days).

**Reply by rickstaa** [reply]
Your correct the pull request only resolved the issue for some users (see https://github.com/anuraghazra/github-readme-stats/issues/2698 and https://github.com/anuraghazra/github-readme-stats/issues/2671).
